tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -1326,3 +1328,52 @@ TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST_RampUp_NoLeap)
         EXPECT_EQ(8760.0, HourOfYear);
     }
 }
+
+TEST_F(EnergyPlusFixture, ScheduleFileDSTtoggleOptionTest)
+{
+    // P. Shrestha - February 2022
+
+    // TODO: create this file with test data in it.
+    // This test will fail until you do that.
+    fs::path scheduleFile = configured_source_directory() / "tst/EnergyPlus/unit/Resources/schedule_file1.csv";
+
+    std::string const idf_objects = delimited_string({"Schedule:File,",
+                                                      "  Test1,                   !- Name",
+                                                      "  ,                        !- Schedule Type Limits Name",
+                                                      "  " + scheduleFile.string() + ",              !- File Name",
+                                                      "  1,                       !- Column Number",
+                                                      "  0,                       !- Rows to Skip at Top",
+                                                      "  8760,                    !- Number of Hours of Data",
+                                                      "  Space,                   !- Column Separator",
+                                                      "  No,                      !- Interpolate to Timestep",
+                                                      "  60,                      !- Minutes per item",
+                                                      "  Yes;                     !- Adjust Schedule for Daylight Savings"});
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    const int sch1idx = GetScheduleIndex(*state, "TEST1");
+    ScheduleManager::ScheduleData &sch1 = state->dataScheduleMgr->Schedule(sch1idx);
+
+    // Checks that the member variable got set correctly.
+    ASSERT_TRUE(sch1.UseDaylightSaving);
+
+    // TODO: Add additional Schedule:File blocks above that test other possibilities to make sure they work right
+    // such as: 1.) Setting to "No"
+    //          2.) Leaving empty (should default to "No")
+    //          3.) Omitting the last field (should default to "No")
+    // Test that each option sets UseDaylightSaving correctly on the each schedule.
+
+    state->dataEnvrn->DSTIndicator = 1; // Tells the simulation that we're in daylight savings
+
+    // TODO: Call LookUpScheduleValue with DSTIndicator on and off and check that the correct hour of data is being returned.
+
+    state->dataEnvrn->DSTIndicator = 0; // Tells the simulation that we're NOT in daylight savings
```

Comment:
Oh and I forgot to put it in the comments as well, but also test to make sure that it's pulling the correct values from the schedule when the simulation is _not_ in daylight savings for both the cases of schedules with and without the "observe DST" switch we put in. So both would not jump forward an hour.